radiography/src/main/java/radiography/ViewFilters.kt

```diff
@@ -0,0 +1,50 @@
+package radiography
+
+import android.view.View
+
+public object ViewFilters {
+
+  /** A [ViewFilter] that matches everything. */
+  @JvmStatic
+  public val All: ViewFilter = viewFilterFor<Any> { true }
+
+  /**
+   * Filters out root views that don't currently have the window focus from the output of
+   * [Radiography.scan].
+   */
+  @JvmField
+  public val FocusedWindowViewFilter: ViewFilter = viewFilterFor<View> { view ->
+    view.parent?.parent != null || view.hasWindowFocus()
+  }
+
+  /**
+   * Filters out views with ids matching [skippedIds] from the output of [Radiography.scan].
+   */
+  @JvmStatic
+  public fun skipIdsViewFilter(vararg skippedIds: Int): ViewFilter = viewFilterFor<View> { view ->
+    val viewId = view.id
+    (viewId == View.NO_ID || skippedIds.isEmpty() || skippedIds.binarySearch(viewId) < 0)
+  }
+
+  /**
+   * Creates a new filter that combines this filter with [otherFilter]
+   */
+  @JvmStatic
+  public infix fun ViewFilter.and(otherFilter: ViewFilter): ViewFilter {
+    val thisFilter = this
+    return object : ViewFilter {
+      override fun matches(view: Any) = thisFilter.matches(view) && otherFilter.matches(view)
+    }
+  }
+
+  /**
+   * Returns a [ViewFilter] that matches any instances of [T] for which [predicate] returns true.
+   */
+  @JvmStatic
+  @PublishedApi internal fun <T : Any> viewFilterFor(
```

Comment:
It looks like this s defined both here and in ViewFilter.java? Did you mean to keep this one?

Reply:
The inline one in `ViewFilter.kt` calls this one. This kills two birds with one stone:

1. The inline one isn't visible to Java callers, but this one is (and vice versa, since this is internal).
2. If the inline function used the `object : TypedViewFilter` syntax directly, the anonymous object declaration would be inlined at every call site and it would create a new concrete class at each one. Delegating to another function means the same anonymous class is used for all inline call sites.

Reply:
Added (non-kdoc) comments for both these cases to point out the difference.